### DISCUSSION TITLE
Preventing caching from pypi

### DISCUSSION THREAD

**Discussion by Nola-W** [devpi/devpi#1060] [Q&A]
Hello, I inherited a project where a small devpi server is used to host just a few packages created by/for my team. We don't necessarily need the server to cache packages from pypi, and I've been trying to figure out how to prevent it from happening to save on space; we have had to increase the size of the vm it's hosted on before. Ideally the devpi server will only have the packages we upload on it, and anything else will be downloaded from pypi.

After doing some digging I found some mentions of the `mirror_use_external_urls` but haven't been able to locate proper documentation on how/where to use it. Does anyone know the proper usage of this setting or where the documentation may be if there is any?

**Comment by fschulze** [comment]
You found the right option. Use ``devpi-client`` to set the option:
1. ``devpi use [url of your index]``
2. ``devpi login [username for that index or root]``
3. ``devpi index mirror_use_external_urls=true``

From then on new packages should be served with the URL of the mirror index.
You can use ``devpi remove [package name]`` to remove existing packages. If you have backups of the server data, then you can also take a look at the files and delete the ``+files/[username]/[indexname]/+f`` folder. For a default install that would be ``+files/root/pypi/+f``. Make sure not to remove anything else (remember the backup).
